server/logger.js

```diff
@@ -0,0 +1,100 @@
+import winston from 'winston';
+import fs from 'fs';
+import { fileURLToPath } from 'url';
+import path from 'path';
+
+const { combine, timestamp, printf, colorize } = winston.format;
```

Comment:
The `updateSocket` function now includes an additional parameter `user_id`, which is a good enhancement for associating user tokens with user IDs. However, ensure that the `user_id` is validated before being used to prevent potential issues with undefined or null values. Consider adding a check to handle such cases. 

```diff
+   if (!user_id) {
+       throw new Error('user_id is required');
+   }
```


    *This comment was auto-generated by **Code Bot** to provide suggestions and improvements.*